lib/Serialization/Deserialization.cpp

```diff
@@ -2528,6 +2528,51 @@ ModuleFile::getDeclCheckedImpl(DeclID DID, Optional<DeclContext *> ForcedContext
         break;
       }
 
+      // SWIFT_ENABLE_TENSORFLOW
+      case decls_block::Differentiable_DECL_ATTR: {
+        AutoDiffMode autodiffMode = AutoDiffMode::Reverse;
+        unsigned autodiffModeValue;
+        uint64_t primalNameId;
+        DeclID primalDeclId;
+        uint64_t adjointNameId;
+        DeclID adjointDeclId;
+        ArrayRef<uint64_t> paramValues;
+
+        serialization::decls_block::DifferentiableDeclAttrLayout::readRecord(
+          scratch, autodiffModeValue, primalNameId, primalDeclId, adjointNameId,
+          adjointDeclId, paramValues);
+        autodiffMode = autodiffModeValue
+          ? AutoDiffMode::Reverse
+          : AutoDiffMode::Forward;
+
+        using FuncSpecifier = DifferentiableAttr::FunctionSpecifier;
+        Optional<FuncSpecifier> primal;
+        FuncDecl *primalDecl = nullptr;
+        if (primalNameId != 0 && primalDeclId != 0) {
+          primal = { getIdentifier(primalNameId), DeclNameLoc() };
+          primalDecl = cast<FuncDecl>(getDecl(primalDeclId));
+        }
+        FuncSpecifier adjoint = { getIdentifier(adjointNameId), DeclNameLoc() };
+        FuncDecl *adjointDecl = cast<FuncDecl>(getDecl(adjointDeclId));
+
+        SmallVector<AutoDiffParameter, 4> parameters;
+        for (auto paramValue : paramValues) {
+          auto parameter = paramValue & 0x01
+            ? AutoDiffParameter::getSelfParameter(SourceLoc())
+            : AutoDiffParameter::getIndexParameter(SourceLoc(), paramValue>>1);
```

Comment:
`paramValue >> 1`

Reply:
Done, minor refactoring to keep things within 80 cols.
I too love spaces around operators.